realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -119,45 +120,53 @@ public boolean isValid() {
 
         @Override
         public Long get() {
-            return value;
+            row.checkIfAttached();
+            return (isNull()) ? null : row.getLong(columnIndex);
```

Comment:
The "hot path" argument doesn't seem very significant.  On the other hand, being able to pull `isNull` back up to the super class totally convinces me.  Fixed.